src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -113,9 +113,13 @@
         Ref("NotEqualToSegment_b"),
         Ref("LikeOperatorSegment"),
     ),
+    TemporaryGrammar=Sequence(
+        Sequence("GLOBAL", optional=True),
+        OneOf("TEMP", "TEMPORARY", optional=True),
```

Comment:
I don't think this should be optional or you could have `GLOBAL` without `TEMP` or `TEMPORARY`.

```suggestion
        OneOf("TEMP", "TEMPORARY"),
```

Plus the whole block should be wrapped in an optional when called.

Reply:
Thanks for the review and suggestions!